Fix liveness probe endpoint

tigrato opened the pull request:
Fixes #13207

tigrato committed:
fix liveness probe endpoint

tigrato committed:
fix helm unittests

r0mant commented on examples/chart/teleport-kube-agent/templates/statefulset.yaml:
```diff
@@ -139,7 +139,7 @@ spec:
           protocol: TCP
         livenessProbe:
           httpGet:
-            path: /readyz
```
I think this may have been on purpose as I know our healthz endpoint always just returns 200 so not very useful. @webvictim Do you recall by chance if this was done on purpose?

webvictim replied:
I didn't write the `StatefulSet` version of this chart myself. It's possible that it was an oversight or a misunderstanding of the difference between `/healthz` and `/readyz` - but maybe Kevin had other insight I'm unaware of!

I think the issue as raised is correct because the behaviour should be the same between both `Deployment` and `StatefulSet`.

An interesting thing I learnt recently that `/readyz` won't report healthy after a reconnection until _every_ proxy tunnel has been established - so the time to do this is going to be longer the more proxies you have. I think this should only affect reconnections though rather than the initial startup - I don't know how often Kubernetes runs a readiness check after a successful start.

tigrato replied:
@webvictim readiness probes are only to check if the pod should receive traffic. If it shouldn't receive traffic, then the pod IP is removed from the service/load balancer endpoints list. This monitoring is done throughout the life of the pod and it can be unregistered at any time. Besides that, readiness probes do not cause the pod to restart as opposite to liveness probes.

 In our use case, it is not really important since it's the agent who connects to the proxy and not the opposite

r0mant reviewed (commented):
@tigrato I guess the main question is, do you know why the readiness endpoint takes longer in a StatefulSet vs Deployment? It does not make much sense, maybe there's another issue here?

Dhruvan1217 commented:
> @tigrato I guess the main question is, do you know why the readiness endpoint takes longer in a StatefulSet vs Deployment? It does not make much sense, maybe there's another issue here?

Per my understanding,

@r0mant I think what @tigrato did seems correct i.e to place correct endpoints for readiness and liveliness in the statefulset. The liveliness probe for deployment is already configured with right endpoints i.e /healthz for liveliness and /readyz for readiness. 
Now, the problem here is, the /readyz was placed in the liveliness probe in statefulset which has overall failure threshold of  30s only after which the container is asked to terminate and restart.  From the logs itself it feels like the agent container takes atleast greater than 30s to start and become ready which is why I feel it's going into a restart loop. Same would be the case if deployment is configured with /readyz endpoint in the liveliness probe. 

I hope this helps.

tigrato commented:
> @tigrato I guess the main question is, do you know why the readiness endpoint takes longer in a StatefulSet vs Deployment? It does not make much sense, maybe there's another issue here?

As @Dhruvan1217 said the difference between Statefulset and Deployment is that in the Deployment the liveness probe is already using the `/healthz` endpoint. That's the reason why Deployment is successful although the readiness probe takes exactly the same time. 

https://github.com/gravitational/teleport/blob/c8af1768283a779300bca2f736317cb54e3ac0d9/examples/chart/teleport-kube-agent/templates/deployment.yaml#L143

Even if it was desired to use the `/readz` as a liveness probe, there is a discrepancy between the Deployment configuration and Statefulset.

Regarding the reason why it takes so much time to connect to the cluster, I think it's related to the fact of my local cluster uses a CA that is only valid locally. While trying to connect to auth server directly, the agent fails and fallbacks to a tunnel connection that is successful to establish but it fails when calling the `Ping` method (probably related to DNS taking some time to reply) . After a while the connection is successful. 

<details>
  <summary>Click to expand!</summary>
  
```
2022-06-07T09:38:51Z INFO             Generating new host UUID: 46bf9ab4-4db2-4c20-b86e-8a8b4fb6df32. service/service.go:687
2022-06-07T09:38:51Z DEBU [SQLITE]    Connected to: file:%2Fvar%2Flib%2Fteleport%2Fproc%2Fsqlite.db?_busy_timeout=10000&_sync=FULL&_txlock=immediate, poll stream period: 1s lite/lite.go:246
2022-06-07T09:38:51Z DEBU [SQLITE]    journal_mode=delete, synchronous=2, busy_timeout=10000 lite/lite.go:293
2022-06-07T09:38:51Z DEBU [PROC:1]    Adding service to supervisor. service:readyz.monitor service/supervisor.go:201
2022-06-07T09:38:51Z INFO [PROC:1]    Service diag is creating new listener on 0.0.0.0:3000. service/signals.go:213
2022-06-07T09:38:51Z INFO [DIAG:1]    Starting diagnostic service on 0.0.0.0:3000. service/service.go:2327
2022-06-07T09:38:51Z DEBU [PROC:1]    Adding service to supervisor. service:diagnostic.service service/supervisor.go:201
2022-06-07T09:38:51Z DEBU [PROC:1]    Adding service to supervisor. service:diagnostic.shutdown service/supervisor.go:201
2022-06-07T09:38:51Z DEBU [KEYGEN]    SSH cert authority started with no keys pre-compute. native/native.go:106
2022-06-07T09:38:51Z DEBU [PROC:1]    Adding service to supervisor. service:register.kube service/supervisor.go:201
2022-06-07T09:38:51Z DEBU [PROC:1]    Adding service to supervisor. service:kube.init service/supervisor.go:201
2022-06-07T09:38:51Z DEBU [PROC:1]    Adding service to supervisor. service:common.rotate service/supervisor.go:201
2022-06-07T09:38:51Z DEBU [PROC:1]    Service has started. service:readyz.monitor service/supervisor.go:262
2022-06-07T09:38:51Z DEBU [PROC:1]    Service has started. service:diagnostic.shutdown service/supervisor.go:262
2022-06-07T09:38:51Z DEBU [PROC:1]    Service has started. service:register.kube service/supervisor.go:262
2022-06-07T09:38:51Z DEBU [PROC:1]    Service has started. service:diagnostic.service service/supervisor.go:262
2022-06-07T09:38:51Z DEBU [PROC:1]    No signal pipe to import, must be first Teleport process. service/service.go:909
2022-06-07T09:38:51Z DEBU [PROC:1]    Service has started. service:kube.init service/supervisor.go:262
2022-06-07T09:38:51Z DEBU [PROC:1]    Service has started. service:common.rotate service/supervisor.go:262
2022-06-07T09:38:51Z INFO [PROC:1]    Joining the cluster with a secure token. service/connect.go:399
2022-06-07T09:38:51Z DEBU [PROC:1]    Generating new key pair for Kube first-time-connect. service/connect.go:302
2022-06-07T09:38:52Z DEBU [AUTH]      Registering node to the cluster. auth-servers:[{tele.teleport.local:3080 tcp }] auth/register.go:153
2022-06-07T09:38:52Z DEBU [AUTH]      The first specified auth server appears to be a proxy. auth/register.go:167
2022-06-07T09:38:52Z INFO [AUTH]      Attempting registration via proxy server. auth/register.go:173
2022-06-07T09:38:52Z DEBU [CLIENT]    HTTPS client init(proxyAddr=tele.teleport.local:3080, insecure=true) client/weblogin.go:232
WARNING: You are using insecure connection to SSH proxy https://tele.teleport.local:3080
2022-06-07T09:38:52Z DEBU [CLIENT]    Attempting https://tele.teleport.local:3080/v1/webapi/host/credentials client/https_client.go:92
2022-06-07T09:38:58Z INFO [AUTH]      Successfully registered via proxy server. auth/register.go:180
2022-06-07T09:38:58Z DEBU [PROC:1]    Deleted generated key pair Kube first-time-connect. service/connect.go:288
2022-06-07T09:38:58Z INFO [PROC:1]    Kube has obtained credentials to connect to the cluster. service/connect.go:433
2022-06-07T09:38:58Z DEBU [PROC:1]    Attempting to connect to Auth Server directly. auth-addrs:[tele.teleport.local:3080] service/connect.go:867
2022-06-07T09:39:04Z DEBU [PROC:1]    Failed to connect to Auth Server directly. auth-addrs:[tele.teleport.local:3080] service/connect.go:874
2022-06-07T09:39:04Z DEBU [PROC:1]    Attempting to discover reverse tunnel address. auth-addrs:[tele.teleport.local:3080] service/connect.go:883
2022-06-07T09:39:04Z DEBU [PROC:1]    Attempting to connect to Auth Server through tunnel. auth-addrs:[tele.teleport.local:3080] service/connect.go:885
2022-06-07T09:39:04Z DEBU             Attempting GET tele.teleport.local:3080/webapi/find webclient/webclient.go:112

2022-06-07T09:39:10Z DEBU             Attempting GET tele.teleport.local:3080/webapi/find webclient/webclient.go:112
2022-06-07T09:39:10Z DEBU             Attempting GET tele.teleport.local:3080/webapi/find webclient/webclient.go:112
2022-06-07T09:39:20Z DEBU [HTTP:PROX] No proxy set in environment, returning direct dialer. proxy/proxy.go:269
2022-06-07T09:39:20Z DEBU [HTTP:PROX] No proxy set in environment, returning direct dialer. proxy/proxy.go:269
2022-06-07T09:39:30Z DEBU [PROC:1]    Connected to Auth Server through tunnel. auth-addrs:[tele.teleport.local:3080] service/connect.go:899
2022-06-07T09:39:30Z INFO [PROC:1]    The process successfully wrote the credentials and state of Kube to the disk. service/connect.go:474
2022-06-07T09:39:30Z DEBU [PROC:1]    Connected client: Identity(Kube, cert(46bf9ab4-4db2-4c20-b86e-8a8b4fb6df32.tele.teleport.local issued by tele.teleport.local:120795101546241818514526927929467691766),trust root(tele.teleport.local:120795101546241818514526927929467691766)) service/connect.go:129
2022-06-07T09:39:30Z DEBU [PROC:1]    Error comparing formats:  is not in dotted-tri format service/connect.go:71
2022-06-07T09:39:30Z DEBU [PROC:1]    Connected client Kube failed to execute test call: connection error: desc = "transport: authentication handshake failed: context deadline exceeded". Node or proxy credentials are out of sync. service/connect.go:80
2022-06-07T09:39:30Z ERRO [PROC:1]    Kube failed to establish connection to cluster: <nil>. service/connect.go:86
2022-06-07T09:39:34Z DEBU [PROC:1]    Retrying connection to auth server after waiting 3.645685168s. service/connect.go:98
2022-06-07T09:39:34Z DEBU [PROC:1]    Connected state: never updated. service/connect.go:145
2022-06-07T09:39:34Z INFO [PROC:1]    Connecting to the cluster tele.teleport.local with TLS client certificate. service/connect.go:164
2022-06-07T09:39:34Z DEBU [PROC:1]    Attempting to connect to Auth Server directly. auth-addrs:[tele.teleport.local:3080] service/connect.go:867
2022-06-07T09:39:39Z DEBU [PROC:1]    Failed to connect to Auth Server directly. auth-addrs:[tele.teleport.local:3080] service/connect.go:874
2022-06-07T09:39:39Z DEBU [PROC:1]    Attempting to discover reverse tunnel address. auth-addrs:[tele.teleport.local:3080] service/connect.go:883
2022-06-07T09:39:39Z DEBU [PROC:1]    Attempting to connect to Auth Server through tunnel. auth-addrs:[tele.teleport.local:3080] service/connect.go:885
2022-06-07T09:39:39Z DEBU             Attempting GET tele.teleport.local:3080/webapi/find webclient/webclient.go:112
2022-06-07T09:39:44Z DEBU             Attempting GET tele.teleport.local:3080/webapi/find webclient/webclient.go:112
2022-06-07T09:39:44Z DEBU             Attempting GET tele.teleport.local:3080/webapi/find webclient/webclient.go:112
2022-06-07T09:39:54Z DEBU [HTTP:PROX] No proxy set in environment, returning direct dialer. proxy/proxy.go:269
2022-06-07T09:39:54Z DEBU [HTTP:PROX] No proxy set in environment, returning direct dialer. proxy/proxy.go:269
2022-06-07T09:39:59Z DEBU [PROC:1]    Connected to Auth Server through tunnel. auth-addrs:[tele.teleport.local:3080] service/connect.go:899
2022-06-07T09:39:59Z DEBU [PROC:1]    Connected client: Identity(Kube, cert(46bf9ab4-4db2-4c20-b86e-8a8b4fb6df32.tele.teleport.local issued by tele.teleport.local:120795101546241818514526927929467691766),trust root(tele.teleport.local:120795101546241818514526927929467691766)) service/connect.go:129
2022-06-07T09:39:59Z DEBU [PROC:1]    Error comparing formats:  is not in dotted-tri format service/connect.go:71
2022-06-07T09:39:59Z DEBU [PROC:1]    Connected client Kube failed to execute test call: connection error: desc = "transport: authentication handshake failed: context deadline exceeded". Node or proxy credentials are out of sync. service/connect.go:80
2022-06-07T09:39:59Z ERRO [PROC:1]    Kube failed to establish connection to cluster: <nil>. service/connect.go:86
2022-06-07T09:40:11Z DEBU [PROC:1]    Retrying connection to auth server after waiting 11.721322964s. service/connect.go:98
2022-06-07T09:40:11Z DEBU [PROC:1]    Connected state: never updated. service/connect.go:145
2022-06-07T09:40:11Z INFO [PROC:1]    Connecting to the cluster tele.teleport.local with TLS client certificate. service/connect.go:164
2022-06-07T09:40:11Z DEBU [PROC:1]    Attempting to connect to Auth Server directly. auth-addrs:[tele.teleport.local:3080] service/connect.go:867
2022-06-07T09:40:17Z DEBU [PROC:1]    Failed to connect to Auth Server directly. auth-addrs:[tele.teleport.local:3080] service/connect.go:874
2022-06-07T09:40:17Z DEBU [PROC:1]    Attempting to discover reverse tunnel address. auth-addrs:[tele.teleport.local:3080] service/connect.go:883
2022-06-07T09:40:17Z DEBU [PROC:1]    Attempting to connect to Auth Server through tunnel. auth-addrs:[tele.teleport.local:3080] service/connect.go:885
2022-06-07T09:40:17Z DEBU             Attempting GET tele.teleport.local:3080/webapi/find webclient/webclient.go:112
2022-06-07T09:40:22Z DEBU             Attempting GET tele.teleport.local:3080/webapi/find webclient/webclient.go:112
2022-06-07T09:40:22Z DEBU             Attempting GET tele.teleport.local:3080/webapi/find webclient/webclient.go:112
2022-06-07T09:40:32Z DEBU [HTTP:PROX] No proxy set in environment, returning direct dialer. proxy/proxy.go:269
2022-06-07T09:40:32Z DEBU [HTTP:PROX] No proxy set in environment, returning direct dialer. proxy/proxy.go:269
2022-06-07T09:40:37Z DEBU [PROC:1]    Connected to Auth Server through tunnel. auth-addrs:[tele.teleport.local:3080] service/connect.go:899
2022-06-07T09:40:37Z DEBU [PROC:1]    Connected client: Identity(Kube, cert(46bf9ab4-4db2-4c20-b86e-8a8b4fb6df32.tele.teleport.local issued by tele.teleport.local:120795101546241818514526927929467691766),trust root(tele.teleport.local:120795101546241818514526927929467691766)) service/connect.go:129
2022-06-07T09:40:37Z DEBU [PROC:1]    Error comparing formats:  is not in dotted-tri format service/connect.go:71
2022-06-07T09:40:37Z DEBU [PROC:1]    Connected client Kube failed to execute test call: connection error: desc = "transport: authentication handshake failed: context deadline exceeded". Node or proxy credentials are out of sync. service/connect.go:80
2022-06-07T09:40:37Z ERRO [PROC:1]    Kube failed to establish connection to cluster: <nil>. service/connect.go:86
2022-06-07T09:41:02Z DEBU [PROC:1]    Retrying connection to auth server after waiting 25.00225097s. service/connect.go:98
2022-06-07T09:41:02Z DEBU [PROC:1]    Connected state: never updated. service/connect.go:145
2022-06-07T09:41:02Z INFO [PROC:1]    Connecting to the cluster tele.teleport.local with TLS client certificate. service/connect.go:164
2022-06-07T09:41:02Z DEBU [PROC:1]    Attempting to connect to Auth Server directly. auth-addrs:[tele.teleport.local:3080] service/connect.go:867
2022-06-07T09:41:07Z DEBU [PROC:1]    Failed to connect to Auth Server directly. auth-addrs:[tele.teleport.local:3080] service/connect.go:874
2022-06-07T09:41:07Z DEBU [PROC:1]    Attempting to discover reverse tunnel address. auth-addrs:[tele.teleport.local:3080] service/connect.go:883
2022-06-07T09:41:07Z DEBU [PROC:1]    Attempting to connect to Auth Server through tunnel. auth-addrs:[tele.teleport.local:3080] service/connect.go:885
2022-06-07T09:41:07Z DEBU             Attempting GET tele.teleport.local:3080/webapi/find webclient/webclient.go:112
2022-06-07T09:41:13Z DEBU             Attempting GET tele.teleport.local:3080/webapi/find webclient/webclient.go:112
2022-06-07T09:41:13Z DEBU             Attempting GET tele.teleport.local:3080/webapi/find webclient/webclient.go:112
2022-06-07T09:41:18Z DEBU [HTTP:PROX] No proxy set in environment, returning direct dialer. proxy/proxy.go:269
2022-06-07T09:41:18Z DEBU [HTTP:PROX] No proxy set in environment, returning direct dialer. proxy/proxy.go:269
2022-06-07T09:41:23Z DEBU [PROC:1]    Connected to Auth Server through tunnel. auth-addrs:[tele.teleport.local:3080] service/connect.go:899
2022-06-07T09:41:23Z DEBU [PROC:1]    Connected client: Identity(Kube, cert(46bf9ab4-4db2-4c20-b86e-8a8b4fb6df32.tele.teleport.local issued by tele.teleport.local:120795101546241818514526927929467691766),trust root(tele.teleport.local:120795101546241818514526927929467691766)) service/connect.go:129
2022-06-07T09:41:23Z INFO [PROC:1]    Kube: features loaded from auth server: Kubernetes:true App:true DB:true OIDC:true SAML:true AccessControls:true AdvancedAccessWorkflows:true HSM:true  service/connect.go:76
2022-06-07T09:41:23Z DEBU [PROC:1]    Broadcasting event. event:KubeIdentity service/supervisor.go:370
2022-06-07T09:41:23Z DEBU [PROC:1]    Service is completed and removed. service:register.kube service/supervisor.go:239
2022-06-07T09:41:23Z DEBU [KUBERNETE] Received event "KubeIdentity". service/kubernetes.go:51
2022-06-07T09:41:23Z DEBU [PROC:1]    Creating in-memory backend for [kubernetes]. service/service.go:1577
2022-06-07T09:41:23Z INFO [KUBERNETE] Cache "kube" first init succeeded. cache/cache.go:715
2022-06-07T09:41:23Z INFO [AUDIT:1]   Creating directory /var/lib/teleport/log. service/service.go:2109
2022-06-07T09:41:23Z INFO [AUDIT:1]   Creating directory /var/lib/teleport/log/upload. service/service.go:2109
2022-06-07T09:41:23Z INFO [AUDIT:1]   Creating directory /var/lib/teleport/log/upload/sessions. service/service.go:2109
2022-06-07T09:41:23Z INFO [AUDIT:1]   Creating directory /var/lib/teleport/log/upload/sessions/default. service/service.go:2109
2022-06-07T09:41:23Z INFO [AUDIT:1]   Creating directory /var/lib/teleport/log. service/service.go:2109
2022-06-07T09:41:23Z INFO [AUDIT:1]   Creating directory /var/lib/teleport/log/upload. service/service.go:2109
2022-06-07T09:41:23Z INFO [AUDIT:1]   Creating directory /var/lib/teleport/log/upload/streaming. service/service.go:2109
2022-06-07T09:41:23Z INFO [AUDIT:1]   Creating directory /var/lib/teleport/log/upload/streaming/default. service/service.go:2109
2022-06-07T09:41:23Z DEBU [PROC:1]    Adding service to supervisor. service:fileuploader.service service/supervisor.go:201
2022-06-07T09:41:23Z DEBU [PROC:1]    Adding service to supervisor. service:fileuploader.shutdown service/supervisor.go:201
2022-06-07T09:41:23Z DEBU [PROC:1]    Service has started. service:fileuploader.service service/supervisor.go:262
2022-06-07T09:41:23Z DEBU [PROC:1]    Service has started. service:fileuploader.shutdown service/supervisor.go:262
2022-06-07T09:41:23Z DEBU [PROXY:AGE] Starting agent pool 46bf9ab4-4db2-4c20-b86e-8a8b4fb6df32.tele.teleport.local.tele.teleport.local... cluster:tele.teleport.local reversetunnel/agentpool.go:164
2022-06-07T09:41:23Z INFO [KUBERNETE] Started reverse tunnel client. service/kubernetes.go:159
2022-06-07T09:41:23Z DEBU             Attempting GET tele.teleport.local:3080/webapi/find webclient/webclient.go:112
2022-06-07T09:41:23Z DEBU [KUBERNETE] Starting watch. resource-kind:lock services/watcher.go:182
2022-06-07T09:41:23Z DEBU [KUBERNETE] Reading Kubernetes creds. kubeClusterName:testing kubeconfigPath: serviceType:0 proxy/auth.go:91
2022-06-07T09:41:23Z DEBU [KUBERNETE] Checking Kubernetes impersonation permissions. cluster:testing proxy/auth.go:146
2022-06-07T09:41:23Z DEBU [KUBERNETE] Have all necessary Kubernetes impersonation permissions. cluster:testing proxy/auth.go:166
2022-06-07T09:41:23Z DEBU [KUBERNETE] Initialized Kubernetes credentials cluster:testing proxy/auth.go:189
2022-06-07T09:41:23Z DEBU             Starting kubernetes_service heartbeats for "kubernetes" proxy/server.go:140
2022-06-07T09:41:23Z DEBU [KUBERNETE] Starting Kube heartbeat with announce period: 1m0s, keep-alive period 5m48.668967121s, poll period: 5s srv/heartbeat.go:155
2022-06-07T09:41:23Z DEBU [PROC:1]    Adding service to supervisor. service:kube.serve service/supervisor.go:201
2022-06-07T09:41:23Z DEBU [PROC:1]    Adding service to supervisor. service:kube.shutdown service/supervisor.go:201
2022-06-07T09:41:23Z DEBU [PROC:1]    Service is completed and removed. service:kube.init service/supervisor.go:239
2022-06-07T09:41:23Z DEBU [PROC:1]    Service has started. service:kube.shutdown service/supervisor.go:262
2022-06-07T09:41:23Z DEBU [PROC:1]    Service has started. service:kube.serve service/supervisor.go:262
2022-06-07T09:41:23Z INFO [KUBERNETE] Starting Kube service via proxy reverse tunnel. service/kubernetes.go:264
2022-06-07T09:41:23Z INFO [KUBERNETE] Kubernetes service 9.2.1:v9.2.1-0-gc6daeda62 is starting via proxy reverse tunnel. utils/cli.go:275
2022-06-07T09:41:23Z DEBU [PROC:1]    Broadcasting event. event:KubernetesReady service/supervisor.go:370
2022-06-07T09:41:23Z DEBU [PROC:1]    Broadcasting mapped event. in:KubernetesReady out:EventMapping(in=[KubernetesReady], out=TeleportReady) service/supervisor.go:395
2022-06-07T09:41:23Z INFO [PROC:1]    The new service has started successfully. Starting syncing rotation status with period 10m0s. service/connect.go:486
2022-06-07T09:41:23Z DEBU [PROC:1]    Teleport component "kubernetes" has started. service/state.go:115
2022-06-07T09:41:33Z DEBU [PROXY:AGE] Adding agent(leaseID=1,state=connecting) -> tele.teleport.local:tele.teleport.local:3080. cluster:tele.teleport.local reversetunnel/agentpool.go:308
2022-06-07T09:41:33Z DEBU             Attempting GET tele.teleport.local:3080/webapi/find webclient/webclient.go:112
2022-06-07T09:41:38Z DEBU [HTTP:PROX] No proxy set in environment, returning direct dialer. proxy/proxy.go:269
2022-06-07T09:41:43Z INFO [KUBERNETE] Connected. addr:172.17.0.3:56140 remote-addr:192.168.1.236:3080 leaseID:1 target:tele.teleport.local:3080 reversetunnel/agent.go:418
2022-06-07T09:41:43Z DEBU [KUBERNETE] Agent connected to proxy: [fc4d0ec4-5488-4952-80b8-bd4e989de028.tele.teleport.local fc4d0ec4-5488-4952-80b8-bd4e989de028 tele.teleport.local.tele.teleport.local tele.teleport.local localhost 127.0.0.1 ::1 remote.kube.proxy.teleport.cluster.local kube.teleport.local]. leaseID:1 target:tele.teleport.local:3080 reversetunnel/agent.go:423
2022-06-07T09:41:43Z DEBU [KUBERNETE] Changing state connecting -> connected. leaseID:1 target:tele.teleport.local:3080 reversetunnel/agent.go:210
2022-06-07T09:41:43Z DEBU [KUBERNETE] Discovery request channel opened: teleport-discovery. leaseID:1 target:tele.teleport.local:3080 reversetunnel/agent.go:535
2022-06-07T09:41:43Z DEBU [KUBERNETE] handleDiscovery requests channel. leaseID:1 target:tele.teleport.local:3080 reversetunnel/agent.go:553
```
</details>

webvictim reviewed (approved):
LGTM

tigrato committed:
Merge branch 'master' into tigrato/fix-statefulset-chart
